### PR TITLE
Allow loading multiple ENV files

### PR DESCRIPTION
**Pull request by joeyespo**
This is to match [the way Foreman works](http://ddollar.github.io/foreman/#GLOBAL-OPTIONS).

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by rmg** [comment]
@slnode ok to test

File: lib/envs.js [hunk 1]
```diff
@@ -138,6 +139,20 @@ function loadEnvs(path) {
   return env;
 }
 
+function loadEnvs(path) {
+  if (path.indexOf(',') === -1) {
```
Comment (rmg) [inline lib/envs.js]: Is this needed? I think `path.split(',')` below will handle this case as a single item Array and "just work", won't it?

File: lib/envs.js [hunk 2]
```diff
@@ -138,6 +139,20 @@ function loadEnvs(path) {
...
+    return loadEnvsFile(path);
+  }
+
+  // Load multiple ENV files
+  var paths = path.split(',');
```
Comment (rmg) [inline lib/envs.js]: I'm not usually a fan of overly terse code, but in this case.. I think the shorter form _might_ be more readable at a glance.

```js
return path.split(',').map(loadEnvsFile).reduce(util._extend, {});
```

**Comment by rmg** [comment]
@joeyespo thanks for the pull request!

Is there any chance you could update the help message to indicate the added support for a comma separated list? And address my comments, if you don't mind.

File: lib/envs.js (hunk 1 above)
Comment (joeyespo) [inline lib/envs.js]: You're correct. Was keeping the existing behavior when there's no `','` for a slight performance gain. It's negligible though, so I can remove it if that's preferred.

File: lib/envs.js (hunk 2 above)
Comment (joeyespo) [inline lib/envs.js]: Either one works for me. Want me to make this change?

Comment (rmg) [inline lib/envs.js]: Yes please :-)

Comment (joeyespo) [inline lib/envs.js]: Done in 2fed7aa6e8ebff38e4f6f4fc71f89aad890c4280.

**Comment by joeyespo** [comment]
@rmg Sure, will do!

**Comment by joeyespo** [comment]
@rmg Updated in 85901d11e21878b009fe0902909d66e92b407396.

**Comment by joeyespo** [comment]
@rmg Any idea what's up with these test failures? My initial commit passed, now there's a bunch of failed linting errors from what I could see. Looks like they might be from generated `.js` in `test/`.

**Comment by rmg** [comment]
It's the env tests, all the keys are now in a different order than the fixtures. Hm...

**Comment by joeyespo** [comment]
@rmg Closing and re-opening to see if a fresh PR will fix the tests.

**Comment by joeyespo** [comment]
@rmg #78 is the new PR. Looks like that didn't help though, they failed right away.

Anything you need me to do?

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by rmg** [comment]
@slnode ok to test

**Comment by rmg** [comment]
@joeyespo do the tests pass for you locally when you run `npm test`?

**Comment by joeyespo** [comment]
@rmg On Windows, so the shell scripts aren't working unfortunately. I've been using `npm link` to test it locally.

**Comment by joeyespo** [comment]
@rmg Just ran `npm test`:

```batch
node-foreman>npm run test

> foreman@1.4.1 test node-foreman
> tap test

total ................................................... 0/0

ok
```

**Comment by rmg** [comment]
@joeyespo I just merged #79 to hopefully help with the tests a little bit. Could you try rebasing your branch on the current master and seeing if the tests work any better for you?

**Comment by joeyespo** [comment]
@rmg Done.

**Comment by rmg** [comment]
@joeyespo do the tests give you anything more useful when you run them locally now? You'll need to do `npm update` to get the newer `tap` release.

**Comment by joeyespo** [comment]
@rmg Getting the following output now, both on `master` and on `multiple-env-files`. (I pulled latest and ran `npm update` first.)

```bash
bash-4.3$ npm run test

> foreman@1.4.1 pretest node-foreman
> jshint *.js lib/*.js

ERROR: Can't open *.js
ERROR: Can't open lib/*.js

> foreman@1.4.1 test node-foreman
> tap test/*.test.*

ok test/console-output.test.js .......................... 1/1
ok test/console-trim.test.js ............................ 1/1
ok test/envs-commented.test.js .......................... 1/1
ok test/envs-empty.test.js .............................. 1/1
ok test/envs-json.test.js ............................... 1/1
ok test/envs-quoted.test.js ............................. 1/1

assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: false == true
    at process.<anonymous> (node-foreman\test\run-once.test.js:29:12)
    at emitOne (events.js:77:13)
    at process.emit (events.js:169:7)
not ok test/run-once.test.js ............................ 0/1
    Command: "C:\Program Files\nodejs\node.exe run-once.test.js"
    TAP version 13
    not ok 1 test/run-once.test.js
      ---
        exit:    1
        stderr:  |
          assert.js:89
            throw new assert.AssertionError({
            ^
          AssertionError: false == true
              at process.<anonymous> (node-foreman\test\run-once.test.js:29:12)
              at emitOne (events.js:77:13)
              at process.emit (events.js:169:7)
        command: |
          "C:\Program Files\nodejs\node.exe run-once.test.js"
      ...

    1..1
    # tests 1
    # fail  1

total ................................................... 6/7

not ok

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "test"
npm ERR! node v4.1.0
npm ERR! npm  v2.14.3
npm ERR! code ELIFECYCLE
npm ERR! foreman@1.4.1 test: `tap test/*.test.*`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the foreman@1.4.1 test script 'tap test/*.test.*'.
npm ERR! This is most likely a problem with the foreman package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tap test/*.test.*
npm ERR! You can get their info via:
npm ERR!     npm owner ls foreman
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     node-foreman\npm-debug.log
```

On Windows though, so I had to run within git's `bash`. I noticed that `once` starts a process in a cross-platform way, but the test doesn't. So changing `$FILENAME` to `%FILENAME%"` produces:

```
bash-4.3$ npm run test

> foreman@1.4.1 pretest node-foreman
> jshint *.js lib/*.js

ERROR: Can't open *.js
ERROR: Can't open lib/*.js

> foreman@1.4.1 test node-foreman
> tap test/*.test.*

ok test/console-output.test.js .......................... 1/1
ok test/console-trim.test.js ............................ 1/1
ok test/envs-commented.test.js .......................... 1/1
ok test/envs-empty.test.js .............................. 1/1
ok test/envs-json.test.js ............................... 1/1
ok test/envs-quoted.test.js ............................. 1/1
ok test/run-once.test.js ................................ 1/1
total ................................................... 7/7

ok
```

So yeah, seems like the tests are passing.

While I'm at it, want me to make `run-once.test.js` cross-platform (when run within `bash`) in a separate PR?

**Comment by rmg** [comment]
Ah, right.. Windows wouldn't recognize all the tests that are actually shell scripts. Might need to port those to node to make them more portable.

**Comment by joeyespo** [comment]
@rmg Let me know what else you need from me.

**Comment by glennr** [comment]
+1 for this feature

**Comment by glennr** [comment]
nevermind - I see this works if you supply the -e flag to the nf command directly.

e.g.
``` 
nf -e .env.test start -s
```

Nice one!

**Pull request closed without merging**